xpf.go

```diff
@@ -0,0 +1,80 @@
+package dns
+
+import (
+	"fmt"
+	"net"
+)
+
+// XPF RR. See https://tools.ietf.org/html/draft-bellis-dnsop-xpf-04
+type XPF struct {
+	Hdr  RR_Header
+	Data XPF_Data `dns:"xpf"` // This is seperated out to make this more compatible with the code generation
+}
+
+// XPF_Data encapsulates the xpf data for custom deserialization
+type XPF_Data struct {
+	IpVersion   uint8
+	Protocol    uint8
+	SrcAddress  net.IP
+	DestAddress net.IP
+	SrcPort     uint16
+	DestPort    uint16
+}
+
+// Equals does a deep equals for the XPF Data
+func (xpf_data *XPF_Data) Equals(xpf_data2 XPF_Data) bool {
+	if xpf_data.IpVersion != xpf_data2.IpVersion {
+		return false
+	}
+	if xpf_data.Protocol != xpf_data2.Protocol {
+		return false
+	}
+	if !xpf_data.SrcAddress.Equal(xpf_data2.SrcAddress) {
+		return false
+	}
+	if !xpf_data.DestAddress.Equal(xpf_data2.DestAddress) {
+		return false
+	}
+	if xpf_data.SrcPort != xpf_data2.SrcPort {
+		return false
+	}
+	if xpf_data.DestPort != xpf_data2.DestPort {
+		return false
+	}
+	return true
+}
+
+func (rr *XPF) String() string {
+	return fmt.Sprintf("%v Source=%v:%v Destination=%v:%v", rr.Hdr.String(), rr.Data.SrcAddress, rr.Data.SrcPort, rr.Data.DestAddress, rr.Data.DestPort)
```

Comment:
This should be consistent with `NULL`, `TSIG` and `OPT`. It should output something prefixed by a comment so it can be passed back into the `ZoneParser` (`NewRR`) without causing an error.